app/components/OutcomeBar.tsx

```diff
@@ -95,32 +99,63 @@ export const OutcomeBar = ({ market }: OutcomeBarProps) => {
     market.outcomeTokenMarginalPrices?.[1] ?? lastTradeMarginalPrices?.[1]
   );
 
-  const hasOutcomePercentages = outcome0.percentage && outcome1.percentage;
+  const getOutcomesPercentages = () => {
+    if (isAnswerFinal && winnerOutcome) {
+      return winnerOutcome.index ? ['0', '100'] : ['100', '0'];
+    }
+
+    return [outcome0?.percentage, outcome1?.percentage];
+  };
+
+  const outcomesPercentages = getOutcomesPercentages();
+  const outcome0percentage = outcomesPercentages[0];
+  const outcome1percentage = outcomesPercentages[1];
+  const hasOutcomePercentages = outcome0percentage && outcome1percentage;
+  const showOutcome0 = !!Number(outcome0percentage) && isAnswerFinal;
+  const showOutcome1 = !!Number(outcome1percentage) && isAnswerFinal;
 
   return (
     <div className="w-full space-y-1">
       <div className="flex space-x-1 transition-all">
-        <div
-          className="flex h-3 items-center rounded-s-8 bg-surface-success-accent-2 px-2"
-          style={{
-            width: outcome0?.percentage ? `${outcome0.percentage}%` : '50%',
-          }}
-        />
-
-        <div
-          className="flex h-3 items-center rounded-e-8 bg-surface-danger-accent-2 px-2"
-          style={{
-            width: outcome1?.percentage ? `${outcome1.percentage}%` : '50%',
-          }}
-        />
+        {showOutcome0 && (
+          <div
+            className={cx(
+              'flex h-3 items-center rounded-s-8 bg-surface-success-accent-2 px-2',
+              {
+                'rounded-e-8': !showOutcome1,
+              }
```

Comment:
I don't think we need the braces, might be wrong tho, would be cleaner if not.
```suggestion
                'rounded-e-8': !showOutcome1,
```

Reply:
`cx` is basically a `clsx` extension, in the CVA docs they suggest to check `clsx` usage:
https://github.com/lukeed/clsx?tab=readme-ov-file#usage

I think there's a way to use a shortcircuit instead the braces.